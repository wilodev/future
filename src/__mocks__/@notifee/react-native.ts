export const { EventType } = jest.requireActual('@notifee/react-native/dist/types/Notification');
export const { AuthorizationStatus } = jest.requireActual(
  '@notifee/react-native/dist/types/Notification',
);
export const { RepeatFrequency, TriggerType } = jest.requireActual(
  '@notifee/react-native/dist/types/Trigger',
);

export default {
  cancelTriggerNotifications: jest.fn(),
  createChannel: jest.fn(),
  createTrigger: jest.fn(),
  createTriggerNotification: jest.fn(),
  getNotificationSettings: jest.fn(),
  getTriggerNotificationIds: jest.fn(),
  onBackgroundEvent: jest.fn(),
  onForegroundEvent: jest.fn(),
  requestPermission: jest.fn(),
};
