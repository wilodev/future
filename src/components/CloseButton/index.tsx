import { StackNavigationProp } from '@react-navigation/stack';
import React from 'react';
import { Image, Platform, Pressable, StyleSheet } from 'react-native';

import { Theme, useTheme } from '~/designSystem/themes';
import { RootNavigationParamList } from '~/navigators/RootNavigationStack';

type CloseButtonProps = {
  navigation: StackNavigationProp<RootNavigationParamList>;
};

/**
 * CloseButton component.
 *
 * @param {Object} navigation - The navigation object.
 * @returns {JSX.Element} - The CloseButton component.
 */
export default function CloseButton({ navigation }: CloseButtonProps): JSX.Element {
  const theme = useTheme();
  const styles = createStyleSheet(theme);

  /**
   * Handles the onPress event of the CloseButton component.
   */
  const handlePress = (): void => {
    navigation.goBack();
  };
  return (
    <Pressable onPress={handlePress} testID="close-button">
      <Image style={styles.closeButton} source={require('~/assets/cross.png')} />
    </Pressable>
  );
}

const createStyleSheet = ({ colors }: Theme) =>
  StyleSheet.create({
    closeButton: {
      tintColor: colors.text,
      marginRight: Platform.select({ ios: 15 }),
    },
  });
