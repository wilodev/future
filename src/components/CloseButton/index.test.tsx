import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import CloseButton from '.';

// Mocking the navigation prop
const mockNavigation: any = {
  goBack: jest.fn(),
};

describe('Close Button', () => {
  // Tests that CloseButton component renders without crashing
  it('test_renders_without_crashing', () => {
    const tree = render(<CloseButton navigation={mockNavigation} />).toJSON();
    expect(tree).toMatchSnapshot(`--updateSnapshot`);
  });
  // Tests that CloseButton component calls navigation.goBack() when pressed
  it('test_press_close_button', () => {
    const { getByTestId } = render(<CloseButton navigation={mockNavigation} />);
    // Find the pressable component by testID
    const pressable = getByTestId('close-button');
    // Simulate a press event on the pressable component
    fireEvent.press(pressable);
    // Check if navigation.goBack() has been called
    expect(mockNavigation.goBack).toHaveBeenCalled();
  });
});
