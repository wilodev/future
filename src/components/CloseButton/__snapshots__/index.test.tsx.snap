// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Close Button test_renders_without_crashing: --updateSnapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="close-button"
>
  <Image
    source={
      {
        "uri": "fake/image.png",
      }
    }
    style={
      {
        "marginRight": 15,
        "tintColor": "rgb(28, 28, 30)",
      }
    }
  />
</View>
`;
